[view-transitions] Incorrect position when view transition is partially scrolled off viewport.

mattwoodrow opened the pull request:
#### 3a5477d552d99df32f465dd33b9725df8c49e283
<pre>
[view-transitions] Incorrect position when view transition is partially scrolled off viewport.
<a href="https://bugs.webkit.org/show_bug.cgi?id=270147">https://bugs.webkit.org/show_bug.cgi?id=270147</a>
&lt;<a href="https://rdar.apple.com/123668892">rdar://123668892</a>&gt;

Reviewed by Tim Nguyen.

The ::view-transition-group pseudo-element is position:fixed, and thus attached to the initial
containing block. The computed transfrom from the old element to it&apos;s position as a child of this
pseudo needs to include the scroll position of the RenderView.

* LayoutTests/TestExpectations:
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::ViewTransition::copyElementBaseProperties):

Canonical link: <a href="https://commits.webkit.org/277371@main">https://commits.webkit.org/277371@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/818445f115d529d868fc0b385d2aa02ba51590a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47374 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26554 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50026 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50057 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43422 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49681 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32071 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24014 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38570 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47955 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24089 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40820 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19890 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21527 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41986 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5417 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/43727 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42398 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51933 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22404 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18741 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45867 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23679 "Built successfully") | | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44897 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/proxy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10464 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24465 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23398 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/dom/ViewTransition.cpp:
```diff
@@ -634,6 +634,10 @@ Ref<MutableStyleProperties> ViewTransition::copyElementBaseProperties(Element& e
         if (!container)
             break;
         LayoutSize containerOffset = renderer->offsetFromContainer(*container, LayoutPoint());
+        if (container->isRenderView()) {
+            auto frameView = renderer->view().protectedFrameView();
+            containerOffset -= toLayoutSize(frameView->scrollPositionRespectingCustomFixedPosition());
+        }
```
Does it work within scrollers?

mattwoodrow replied:
It does! Just the FrameView scrolling is special.

mattwoodrow committed:
[view-transitions] Incorrect position when view transition is partially scrolled off viewport.
https://bugs.webkit.org/show_bug.cgi?id=270147
<rdar://123668892>

Reviewed by Tim Nguyen.

The ::view-transition-group pseudo-element is position:fixed, and thus attached to the initial
containing block. The computed transfrom from the old element to it's position as a child of this
pseudo needs to include the scroll position of the RenderView.

* LayoutTests/TestExpectations:
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::ViewTransition::copyElementBaseProperties):

Canonical link: https://commits.webkit.org/277371@main
